pilot/pkg/networking/core/httproute.go

```diff
@@ -257,7 +257,7 @@ func selectVirtualServices(virtualServices []config.Config, servicesByName map[h
 			// TODO: This is a bug. VirtualServices can have many hosts
 			// while the user might be importing only a single host
 			// We need to generate a new VirtualService with just the matched host
-			if servicesByName[host.Name(h)] != nil {
+			if servicesByName[host.Name(strings.ToLower(h))] != nil {
 				match = true
 				break
 			}
```

Comment:
If we want hosts to be case insensitive, then we should do the same for wildcard matches (httproute.go:264-283). This still fails if I have a vs with host `*.facebook.COM`.

I feel like this solution is fragile because it relies on calling `strings.ToLower` in all the right places and wonder if it would be better to have a more abstract solution (say a proper `Hostname` type).

Also, more thorough testing would probably be good.

Reply:
I think wildcard match is a different problem - should not be tied to this. Not sure if we want to support that model. Ideally I think we should have a warning to not have mixed case in virtual hosts. 
@hzxuzhonghu @keithmattix  WDYT?

Regarding generic solution, I think that needs more thought. If we agree on fix this, my suggestion would be merge this and think about generic solution

Reply:
I'm wondering why wildcard matching isn't as simple as just calling Strings.Lower on the pattern as well.

Reply:
I was thinking of the risk with changing general wildcard matcher. But I found a simpler way to handle this for this case. PTAL

Reply:
Oh I see, you meant adding the casing logic to `Name.Matches` might be risky, but if we lowercase it before we call it, it should be fine.